main_test.py

```diff
@@ -19,11 +19,18 @@
 
 import base64
```

Comment:
should we add a unicode testcase, since there is some decoding happening in main?

Reply:
Good point, I also added some other test cases as well to be more comprehensive.